sched/scheduler/task_runner.go

```diff
@@ -29,33 +31,65 @@ func runTaskAndLog(
 	task sched.TaskDefinition,
 	markCompleteOnFailure bool,
 	stat stats.StatsReceiver) error {
-	// Log StartTask Message to SagaLog
-	err := saga.StartTask(taskId, nil)
+
+	// Start Task and then log StartTask message a status payload containing stdout/stderr URIs.
+	// Note: this ordering will result in an orphaned worker if logging fails.
+	//       also, URIs will likely change upon task completion, i.e. from local file to snapshot URI.
+	// TODO: !!this no longer works if we ever start allowing compensating actions on failure!!
```

Comment:
We should check with Caitie on this. I'd prefer to not compromise the model. It would probably be easy to add a new message. WDYT?